app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/ViewerFragment.java

```diff
@@ -176,15 +138,74 @@ public void onDetach() {
     @Override
     public void onDestroyView() {
         super.onDestroyView();
-        MessageBroadcastReceiver.unregisterReceiver(getActivity(), mUserLocationReceiver);
-        mLocationFetcher.unregisterReceiver(mLocationReceiver);
+        if (mUserLocationReceiver != null) {
+            MessageBroadcastReceiver.unregisterReceiver(getActivity(), mUserLocationReceiver);
+        }
+        if (mLocationFetcher != null && mLocationReceiver != null) {
+            mLocationFetcher.unregisterReceiver(mLocationReceiver);
+        }
     }
 
-    public void onCreateGeographicMessage(PointGeometry pointGeometry) {
-        SendGeographicMessageDialog sendGeographicMessageDialogFragment =
-                SendGeographicMessageDialog.newInstance(pointGeometry, this);
+    @Override
+    public void drawPin(PointGeometry pointGeometry) {
+        if (pointGeometry == null) {
+            throw new IllegalArgumentException("given pointGeometry is null!");
+        }
 
-        sendGeographicMessageDialogFragment.show(getFragmentManager(), "sendCoordinatesDialog");
+        addPinPoint(pointGeometry, mSentLocationsLayer);
+    }
+
+    @Override
+    public void goToLocation(PointGeometry pointGeometry) {
+        mGGMapView.zoomTo(pointGeometry);
+    }
+
+    public GGMap getGGMap() {
+        return mGGMapView;
+    }
+
+    @Override
+    public void onGGMapViewReady() {
+        //Zoom to Israel~
+        mGGMapView.setExtent(34, 29, 36, 34);
+
+        mGGMapView.addLayer(mSentLocationsLayer);
+        mGGMapView.addLayer(mUsersLocationsLayer);
+
+        registerForLocationUpdates();
+    }
+
+    private void registerForLocationUpdates() {
```

Comment:
what happens with life cycle?
onRestart? onResume? does this function will be called again?